dnsdist doesn't build without dnscrypt enabled

<!-- Hi! Thanks for filing an issue. It will be read with care by human beings. Can we ask you to please fill out this template and not simply demand new features or send in complaints? Thanks! -->
<!-- Also please search the existing issues (both open and closed) to see if your report might be duplicate -->
<!-- Please don't file an issue when you have a support question, send support questions to the mailinglist or ask them on IRC (https://www.powerdns.com/opensource.html) -->

<!-- Tell us what is issue is about -->
 - Program: dnsdist <!-- delete the ones that do not apply -->
 - Issue type: Bug report <!-- delete the one that does not apply -->

### Short description
<!-- Explain in a few sentences what the issue/request is -->
If configured without "--enable-dnscrypt", dnsdist fails to compile.



<!--
If this is a bug report, use the following part of the the template and delete the part at the bottom
-->
### Environment
<!-- Tell us about the environment -->
 - Operating system: NetBSD/amd64-current
 - Software version: HEAD as of 2019-01-11
 - Software source: git <!-- e.g. Operating system repository, PowerDNS repository, compiled yourself -->

### Steps to reproduce
<!-- Tell us step-by-step how the issue can be triggered. Please include your configuration files and any (Lua) scripts that are loaded. -->
1. cd pdns/dnsdistdist <!-- step 1 -->
2. ./configure <!-- step 2 -->
3. gmake <!-- step 3 -->

### Expected behaviour
<!-- What would you expect to happen when the reproduction steps are run -->
dnsdist should be built. :)

### Actual behaviour
<!-- What did happen? Please (if possible) provide logs, output from `dig` and/or tcpdump/wireshark data -->

dnsdist.cc: At global scope:
dnsdist.cc:1323:155: error: 'DNSCryptQuery' was not declared in this scope
 static int sendAndEncryptUDPResponse(LocalHolders& holders, ClientState& cs, const DNSQuestion& dq, char* response, uint16_t responseLen, std::shared_ptr<DNSCryptQuery>& dnsCryptQuery, int delayMsec, const ComboAddress& dest, struct mmsghdr* responsesVect, unsigned int* queuedResponses, struct iovec* respIOV, char* respCBuf, bool cacheHit)

A handful of related error messages follow, mostly complaints that "dnsCryptQuery" is undeclared.


### Other information
<!-- if you already did more digging into the issue, please provide all the information you gathered -->

--- a/pdns/dnscrypt.hh
+++ b/pdns/dnscrypt.hh
@@ -22,7 +22,13 @@
 #pragma once
 #include "config.h"
 
-#ifdef HAVE_DNSCRYPT
+#ifndef HAVE_DNSCRYPT
+
+class DNSCryptQuery
+{
+};
+
+#else
 
 #include <memory>
 #include <string>

--- a/pdns/dnsdist.cc
+++ b/pdns/dnsdist.cc
@@ -1375,9 +1375,9 @@ static void processUDPQuery(ClientState& cs, LocalHolders& holders, const struct
     gettime(&now);
     gettime(&queryRealTime, true);
 
-#ifdef HAVE_DNSCRYPT
     std::shared_ptr<DNSCryptQuery> dnsCryptQuery = nullptr;
 
+#ifdef HAVE_DNSCRYPT
     if (!checkDNSCryptQuery(cs, query, len, dnsCryptQuery, dest, remote, queryRealTime.tv_sec)) {
       return;
     }

